src/app/util/redis.py

```diff
@@ -31,43 +32,17 @@ def set(self, key, value):
 
     def set_complex(self, key, complex_value):
         """ Set a complex key-value element by converting to json string """
-        json_value = standardise(jsons.dump(complex_value))
+        json_value = jsonpickle.encode(complex_value, unpicklable=False)
         self.log.debug("[set_complex] Successful conversion to JSON, setting value: " + json_value)
         return self.client.set(key, json_value)
 
-    def get_complex(self, key):
+    def get_complex(self, key, cls: Base):
         """ Get a complex key-value element by converting from json string """
         json_value = self.client.get(key)
-        if json_value is None:
-            return None
+        if json_value is not None:
+            try:
+                return cls.from_json(json_value)
```

Comment:
@AdamJ335 
This is the main point: json comes in from redis and gets converted into a python object (via a dictionary conversion that you highlighted before)